### PR TITLE
Adding  native implementation for ESP32 RMT library

### PR DESCRIPTION
**Pull request by ivan0414**
## Description
- Adding stub functions for native RMT for ESP32.

## Motivation and Context

- Addresses nanoFramework/Home#518.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/nanoFramework.Hardware.ESP32.RMT/nanoFramework_hardware_esp32_rmt_native_nanoFramework_Hardware_Esp32_Rmt_TransmitterChannel.cpp [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+//    prevent loss of your changes when the tool is
+//    re-run.
+//
+//-----------------------------------------------------------------------------
+#include <vector>
+#include <map>
+#include <cstring>
+#include "nanoFramework_hardware_esp32_rmt_native.h"
+
+#define MY_CLEANUP()							hr = S_OK; nanoCLR_Cleanup: return
```
Comment (josesimoes) [inline targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/nanoFramework.Hardware.ESP32.RMT/nanoFramework_hardware_esp32_rmt_native_nanoFramework_Hardware_Esp32_Rmt_TransmitterChannel.cpp:17]: You should be using the standard macros from the CLR not implementing you own versions.

File: targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/nanoFramework.Hardware.ESP32.RMT/nanoFramework_hardware_esp32_rmt_native_nanoFramework_Hardware_Esp32_Rmt_TransmitterChannel.cpp [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+#define MY_CLEANUP_D(r)							hr = S_OK; nanoCLR_Cleanup: return r
+#define CHANNEL(ch)                             static_cast<rmt_channel_t>(ch)
+
+#define DEFAULT_DEVIDER                         4
+
+static std::map<rmt_channel_t, std::vector<rmt_item32_t>> Library_nanoFramework_hardware_esp32_rmt_native_nanoFramework_Hardware_Esp32_Rmt_TransmitterChannel::registredChannels;
+
+signed int Library_nanoFramework_hardware_esp32_rmt_native_nanoFramework_Hardware_Esp32_Rmt_TransmitterChannel::NativeInit___STATIC__I4__I4(signed int channel, HRESULT &hr)
```
Comment (josesimoes) [inline targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/nanoFramework.Hardware.ESP32.RMT/nanoFramework_hardware_esp32_rmt_native_nanoFramework_Hardware_Esp32_Rmt_TransmitterChannel.cpp:25]: These calls have to return HRESULT, nothing else.

Comment (josesimoes) [reply targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/nanoFramework.Hardware.ESP32.RMT/nanoFramework_hardware_esp32_rmt_native_nanoFramework_Hardware_Esp32_Rmt_TransmitterChannel.cpp:25]: You must have changed what was in the stubs generated automatically. You can't do that. Just "allowed" to touch the code inside, not around...

**Comment by networkfusion** [comment]
It would also be good to compare this against #1442 so that it can be closed...

**Comment by josesimoes** [comment]
@ivan0414 let's put this on hold until we release the next stable version OK?
That shouldn't take long.

**Comment by josesimoes** [comment]
Ignoring style and other checks. To be fixed in the working branch.

**Pull request merged**